e3/anod/checkout.py

```diff
@@ -31,6 +31,33 @@ def __init__(self, name, working_dir):
             os.path.join(working_dir, self.name))
         self.metadata_file = self.working_dir + '_checkout.json'
 
+    def update_fingerprint(self, fingerprint, is_prediction=False):
+        """Given a fingerprint amend it with metadata of the current checkout.
+
+        :param fingerprint: initial fingerprint
+        :type fingerprint: Fingerprint
+        :param is_prediction: if True this is an attempt to compute the
+            fingerprint before doing an update. For a checkout the returned
+            value in that case is always the None.
+            See e3.job.walk.compute_fingerprint for more information.
+        :type is_prediction: bool
+        :return: a fingerprint
+        :rtype: None | Fingerprint
+        """
+        if is_prediction:
+            return None
+
+        try:
+            with open(self.metadata_file) as fd:
+                vcs_info = json.load(fd)
+            fingerprint.add(
+                self.name, (vcs_info['url'],
+                            vcs_info['new_commit']))
+            return fingerprint
+        except Exception:
+            logger.exception('Got exception')
```

Comment:
I think we will find it very useful for analysis purposes to log the exception itself.

Reply:
the call to logger.exception will output the traceback and all the relevant exception information. Regarding the interface, I am returning the fingerprint as in some cases we return None (which is in the end a special fingerprint value). The use case for this code will be to call it from the walker to compute the fingerprint.

Reply:
Re: logger.exception: Ok, good.

Still don't understand all that well the use case, but I suppose this will become clearer when I see it actually become used.

No further comment on this pull request.